lib/widgets/message_list.dart

```diff
@@ -635,3 +702,75 @@ final _kMessageTimestampStyle = TextStyle(
   fontSize: 12,
   fontWeight: FontWeight.w400,
   color: const HSLColor.fromAHSL(0.4, 0, 0, 0.2).toColor());
+
+Future<void> markNarrowAsRead(BuildContext context, ApiConnection connection, Narrow narrow) async {
+  final zulipLocalizations = ZulipLocalizations.of(context);
```

Comment:
This is based on some code we have elsewhere, so links to that code will be helpful for review. We also may want to keep the UX aligned with the one in web, so the comparison may be useful for making future changes too — so the links will be helpful to put in the code itself, rather than the commit message. So like:
```suggestion
Future<void> markNarrowAsRead(BuildContext context, ApiConnection connection, Narrow narrow) async {
  // Compare web's `FUNCTION` in web/src/FILENAME.js
  // and zulip-mobile's `FUNCTION` in src/action-sheets/index.js .
  final zulipLocalizations = ZulipLocalizations.of(context);
```